cmd/krew/cmd/search_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func Test_searchByNameAndDesc(t *testing.T) {
+	keyword := "config"
+	testPlugins := []struct {
+		names    []string
+		descs    []string
+		expected []string
+	}{
+		{
+			names: []string{"switch-config", "ctx", "kc"},
+			descs: []string{
+				" Switches between kubeconfig files",
+				"Switch between contexts in your kubeconfig",
+				"Interactive CRUD operations to manage kubeconfig",
+			},
+			expected: []string{"switch-config", "ctx", "kc"},
```

Comment:
my advice would be to use some abstract descriptions like "foo bar quux foobar" so that we can clearly see which part is it matching (name? desc? is it full match? partial/fuzzy match?)

Reply:
Updated.